api/cmd/portainer/main.go

```diff
@@ -157,6 +157,16 @@ func initDataStore(flags *portainer.CLIFlags, secretKey []byte, fileService port
 	return store
 }
 
+// checkDBSchemaServerVersionMatch checks if the server version matches the db scehma version
+func checkDBSchemaServerVersionMatch(dbStore dataservices.DataStore, serverVersion string) bool {
+	v, err := dbStore.Version().Version()
+	if err != nil {
+		return false
+	}
+
+	return v.SchemaVersion == serverVersion && v.Edition == int(portainer.Edition)
```

Comment:
why is serverVersion a parameter and edition is taken from the global scope? I'd say either take one way or the other, but not a mixed approach

Reply:
Thanks for letting me know. I found that the edition needs to be compared after initial submission and forgot update the function. It is a good catch. I have fixed it in another PR.